Core/include/Acts/Detector/IndexedGridFiller.hpp

```diff
@@ -0,0 +1,385 @@
+struct CenterReferenceGenerator {
+  /// Helper to access the Center point of for filling the grid
+  ///
+  /// @param gctx the geometry context of this operation
+  /// @param surface the surface for which the reference point is to be accessed
+  ///
+  /// @return a vector of referene points for filling
+  const std::vector<Vector3> references(const GeometryContext& gctx,
+                                        const Surface& surface) const {
+    return {surface.center(gctx)};
+  }
+};
+
+/// A struct to access reference postions based on bin values
+///
+/// This generator will provide only one filling point and hence
+/// only a single bin in the indexed grid.
+struct BinningValueReferenceGenerator {
+  /// The binning value
+  BinningValue bValue;
+
+  /// Helper to access a reference postion based on binning value
+  ///
+  /// @param gctx the geometry context of this operation
+  /// @param surface the surface for which the reference point is to be accessed
+  ///
+  /// @return a vector of referene points for filling
+  const std::vector<Vector3> references(const GeometryContext& gctx,
+                                        const Surface& surface) const {
+    return {surface.binningPosition(gctx, bValue)};
+  }
+};
+
+/// A struct to access generated vertices from surface polyhedrons
+/// These vertices are then used to find the bin boundary box for the
+/// indexed grid.
+///
+/// The grid filling then completes the empty bins in between and
+/// expands if necessary.
+struct PolyhedronReferenceGenerator {
+  /// Also use the barycenter
+  bool addBarycenter = true;
+
+  /// The number of segments to approximate (1 means extrema points only)
+  unsigned int nSegments = 1;
+
+  /// Helper to access the Center point of for filling the grid
+  ///
+  /// @param gctx the geometry context of this operation
+  /// @param surface the surface for which the reference point is to be accessed
+  ///
+  /// @return a vector of referene points for filling
+  const std::vector<Vector3> references(const GeometryContext& gctx,
+                                        const Surface& surface) const {
+    // Create the return  vector
+    std::vector<Vector3> rPositions;
+    auto pHedron = surface.polyhedronRepresentation(gctx, nSegments);
+    rPositions.insert(rPositions.end(), pHedron.vertices.begin(),
+                      pHedron.vertices.end());
+    // Add the barycenter if configured
+    if (addBarycenter) {
+      Vector3 bc(0., 0., 0.);
+      std::for_each(rPositions.begin(), rPositions.end(),
+                    [&](const auto& p) { bc += p; });
+      bc *= 1. / rPositions.size();
+      rPositions.push_back(bc);
+    }
+    return rPositions;
+  }
+};
+
+/// A helper class that fills surfaces into predefined grids
+struct IndexedGridFiller {
+  // Bin expansion where needed
+  std::vector<std::size_t> binExpansion = {};
+
+  // An ouput logging level
+  Logging::Level logLevel = Logging::INFO;
```

Comment:
In this specific case, why not just change this to
```cpp
std::shared_ptr<const Logger> logger = getDefaultLogger("IndexedGridFiller", Logging::INFO)
```
or `unique_ptr` if you don't need this to be copyable?

Reply:
Exactly, done like this now.